Replace governance documents with link to cert-manager community documents.

inteon opened the pull request:
This PR completes the move of most of our governance files to the community repo: https://github.com/cert-manager/community/pull/10

1. replace these files with documents that link to the community documents:
   - CODE_OF_CONDUCT.md
   - SECURITY.md
   - USERS.md
   - SECURITY_CONTACTS.md 
2. add a new GOVERNANCE.md document that links to the community document.
3. add clo monitor hint to let it find the license scanner output
4. update README.md:
   - adds clomonitor badge
   - adds direct links to slack repos

### Kind

/kind cleanup

### Release Note

```release-note
NONE
```

maelvls commented:
> Adds MAINTAINERS.md and GOVERNANCE.md files that reference the real files

Why would we add these? Istio and Kubernetes don't have "sym link" to their governance and maitainers files.

> Add CLOMonitor

Why do we need this?

inteon commented:
I'm trying to improve the score that is shown on the CNCF website:
<img width="883" alt="image" src="https://github.com/cert-manager/cert-manager/assets/42113979/aa1da3e4-c885-487d-aa11-c03f88bd6d7c">

Adding the MAINTAINERS.md and GOVERNANCE.md files will make this check succeed: 
<img width="563" alt="image" src="https://github.com/cert-manager/cert-manager/assets/42113979/a27a0786-87f7-4c04-aa52-f656fa236b4e">
(https://clomonitor.io/projects/cncf/cert-manager#cert-manager_documentation)

inteon commented:
waiting for this update: https://github.com/cncf/clomonitor/pull/1270

/hold

inteon commented:
@maelvls blocked by https://github.com/cert-manager/community/pull/10

maelvls commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,3 @@
+# Maintainers
```
```suggestion
# Project Governance
```

maelvls commented:
/lgtm
/approve

inteon commented:
/unhold

inteon commented:
/hold

inteon commented:
@maelvls I updated the PR, so we replace the files that we moved to the community repo with links to their new location. These files were created in the community repo in https://github.com/cert-manager/community/pull/10.

maelvls reviewed (commented):
It looks all good!

/lgtm
/approve
